### PR TITLE
Add is_default field to taxonomy sql models

### PR DESCRIPTION
**Pull request by allisonking**
Closes https://github.com/ethyca/fides/issues/854

This requires a fideslang update first: https://github.com/ethyca/fideslang/pull/68

### Code Changes

* [x] Adds a field `is_default` to the taxonomy related sql models (DataCategory, DataUse, DataSubject, DataQualifier)
* [x] Alembic migration

### Steps to Confirm

* [ ] nox -s dev
* [ ] Check out `localhost:8080/docs` and open up one of the taxonomy fields (i.e. GET data_category)
* [ ] Should also see the new `is_default` field in `localhost:8080/openapi.json`

image

### Pre-Merge Checklist

* [x] All CI Pipelines Succeeded
* Documentation Updated:
  - [ ] documentation complete, or draft/outline provided (tag docs-team to complete/review on this branch)
  - [ ] documentation issue created (tag docs-team to complete issue separately)
* [x] Issue Requirements are Met 
* [x] Relevant Follow-Up Issues Created 
  * [ ] https://github.com/ethyca/fides/issues/986
  * [ ] https://github.com/ethyca/fides/issues/988
* [x] Update `CHANGELOG.md`

### Description Of Changes

The autogenerated alembic migration looks like it's deleting and recreating indexes. Not sure if that's expected, or if I should just remove those lines?

### CODE REVIEW

**Comment by allisonking** [comment]
Not sure what the docs error is about 😕

**Comment by allisonking** [comment]
Added this issue to cover preventing users from modifying the default taxonomy https://github.com/ethyca/fides/issues/986

**Comment by allisonking** [comment]
Bumped fideslang version, should be ready for review now (though docs CI is still failing 😢 )

**Comment by allisonking** [comment]
whoa never mind, docs build passes now!

File: src/fidesctl/api/ctl/migrations/versions/f131a1263a10_add_is_default_to_taxonomy_fields.py [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+        op.f("ix_data_categories_id"), "data_categories", ["id"], unique=False
+    )
+    op.add_column(
+        "data_qualifiers", sa.Column("is_default", sa.BOOLEAN(), nullable=True)
+    )
+    op.drop_index("ix_data_qualifiers_id", table_name="data_qualifiers")
+    op.create_index(
+        op.f("ix_data_qualifiers_id"), "data_qualifiers", ["id"], unique=False
+    )
+    op.add_column("data_subjects", sa.Column("is_default", sa.BOOLEAN(), nullable=True))
```
Comment (ThomasLaPiana) [inline src/fidesctl/api/ctl/migrations/versions/f131a1263a10_add_is_default_to_taxonomy_fields.py:34]: did you ever figure out why the index changes were happening or confirm they existed before?

Weird, but seems harmless enough

**Review by ThomasLaPiana — APPROVED** [review]
Excellent small iteration here! One comment about the alembic migration

File: src/fidesctl/api/ctl/migrations/versions/f131a1263a10_add_is_default_to_taxonomy_fields.py (hunk 1 above)
Comment (allisonking) [reply src/fidesctl/api/ctl/migrations/versions/f131a1263a10_add_is_default_to_taxonomy_fields.py:34]: actually, it looks like they make the id's into `unique=false`, which seems like we wouldn't want that?

old:
```
Indexes:
    "data_categories_pkey" PRIMARY KEY, btree (id)
    "ix_data_categories_fides_key" UNIQUE, btree (fides_key)
    "ix_data_categories_id" UNIQUE, btree (id)
```

new:
```
Indexes:
    "data_categories_pkey" PRIMARY KEY, btree (id)
    "ix_data_categories_fides_key" UNIQUE, btree (fides_key)
    "ix_data_categories_id" btree (id)
```

Comment (allisonking) [reply src/fidesctl/api/ctl/migrations/versions/f131a1263a10_add_is_default_to_taxonomy_fields.py:34]: @sanders41 confirmed this was probably left out of the [migration to change to fideslib](https://github.com/ethyca/fides/blob/main/src/fidesctl/api/ctl/migrations/versions/155fd8e51d9d_add_fideslib_models.py). The behavior is the same, as fideslib added a `primary_key` which is by default already unique, so this should be a safe change 👍

**Pull request merged**